diff --git a/app/code/Mage/Install/Model/Installer/Db/Mysql4.php b/app/code/Mage/Install/Model/Installer/Db/Mysql4.php
index edcf433..b80c982 100644
--- a/app/code/Mage/Install/Model/Installer/Db/Mysql4.php
+++ b/app/code/Mage/Install/Model/Installer/Db/Mysql4.php
@@ -60,6 +60,11 @@ class Mage_Install_Model_Installer_Db_Mysql4 extends Mage_Install_Model_Installe
     {
         $variables  = $this->_getConnection()
             ->fetchPairs('SHOW VARIABLES');
+        // Fix for MySQL 5.6.1+
+        if (!isset($variables['have_innodb'])) {
+             $engines = $this->_getConnection()->fetchPairs('SHOW ENGINES');
+             return (isset($engines['InnoDB']) && ($engines['InnoDB'] == 'DEFAULT' || $engines['InnoDB'] == 'YES'));
+        }
         return (!isset($variables['have_innodb']) || $variables['have_innodb'] != 'YES') ? false : true;
     }
 
